janusgraph-cassandra/src/main/java/org/janusgraph/diskstorage/cassandra/thrift/thriftpool/CTConnectionFactory.java

```diff
@@ -135,8 +143,15 @@ public boolean validateObject(String key, CTConnection c) {
                           c, curCfg, c.getConfig());
             }
         }
+        
+        boolean isOpen = c.isOpen();
+        try {
+            c.getClient().describe_version();
```

Comment:
Sorry but not really, that was the issue before, `isOpen` is always true when network drops the connection, isOpen will always return true. I could remove this check though (line 147) since it is useless.